src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -97,66 +85,51 @@ CHIP_ERROR TransferInit::Pack(System::PacketBuffer * aBuffer)
         if (mWideRange)
         {
             bbuf.PutLE64(mMaxLength);
-            VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
         }
         else
         {
-            bbuf.PutLE32((uint32_t) mMaxLength);
-            VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
+            bbuf.PutLE32(static_cast<uint32_t>(mMaxLength));
         }
     }
 
     VerifyOrExit(mFileDesignator && mFileDesLength > 0, err = CHIP_ERROR_INVALID_ARGUMENT);
     bbuf.PutLE16(mFileDesLength);
-    bbuf.Put(mFileDesignator, (size_t) mFileDesLength);
-    VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
+    bbuf.Put(mFileDesignator, static_cast<size_t>(mFileDesLength));
 
     // Metadata is optional
     if (mMetadata && mMetadataLength > 0)
     {
-        bbuf.Put(mMetadata, (size_t) mMetadataLength);
-        VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
+        bbuf.Put(mMetadata, static_cast<size_t>(mMetadataLength));
     }
 
-    aBuffer->SetDataLength(static_cast<uint16_t>(bbuf.Written()));
+    VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
+    aBuffer.SetDataLength(static_cast<uint16_t>(bbuf.Written()));
 
 exit:
     return err;
 }
 
-CHIP_ERROR TransferInit::Parse(System::PacketBuffer * aBuffer, TransferInit & aParsedMessage)
+CHIP_ERROR TransferInit::Parse(System::PacketBuffer & aBuffer, TransferInit & aParsedMessage)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     uint8_t ptcByte;
     uint8_t rangeCtl;
     bool hasStartOffset     = false;
     uint32_t tmpUint32Value = 0;
```

Comment:
Suggest commenting that this is to possibly read non-wide lengths and offsets.

Reply:
Will do